neo/rawio/wavesurferrawio.py

```diff
@@ -0,0 +1,128 @@
+"""
+In development 16/10/2021
+
+Class for reading data from WaveSurfer, a software written by
+Boaz Mohar and Adam Taylor https://wavesurfer.janelia.org/
+
+Requires the PyWaveSurfer module written by Boaz Mohar and Adam Taylor.
+
+To Discuss:
+- Wavesurfer also has analog output, and digital input / output channels, but here only supported analog input. Is this okay?
+- I believe the signal streams field is configured correctly here, used AxonRawIO as a guide.
+- each segment (sweep) has it's own timestamp, so I beleive no events_signals is correct (similar to winwcprawio not axonrawio)
+
+1) Upload test files (kindly provided by Boaz Mohar and Adam Taylor) to g-node portal
+2) write RawIO and IO tests
+
+2. Step 2: RawIO test:
+* create a file in neo/rawio/tests with the same name with "test_" prefix
+* copy paste neo/rawio/tests/test_examplerawio.py and do the same
+
+4.Step 4 : IO test
+* create a file in neo/test/iotest with the same previous name with "test_" prefix
+* copy/paste from neo/test/iotest/test_exampleio.py
+"""
+
+from .baserawio import (BaseRawIO, _signal_channel_dtype, _signal_stream_dtype,
+                        _spike_channel_dtype, _event_channel_dtype)
+import numpy as np
+
+try:
+    from pywavesurfer import ws
+except ImportError as err:
+    HAS_PYWAVESURFER = False
+    PYWAVESURFER_ERR = err
+else:
+    HAS_PYWAVESURFER = True
+    PYWAVESURFER_ERR = None
+
+class WaveSurferRawIO(BaseRawIO):
+
+    extensions = ['fake']
+    rawmode = 'one-file'
+
+    def __init__(self, filename=''):
+        BaseRawIO.__init__(self)
+        self.filename = filename
+
+        if not HAS_PYWAVESURFER:
+            raise PYWAVESURFER_ERR
+
+    def _source_name(self):
+        return self.filename
+
+    def _parse_header(self):
+
+        pyws_data = ws.loadDataFile(self.filename, format_string="double")
```

Comment:
This is not actually lazy - apologies I understand better now how memmap is working e.g. in the AxonIO, and the logic behind the RawIO / IO API. 

Lazy loading could be supported by re-writing the code but my initial thought was to provide a wrapper around PyWaveSurfers IO only. I see now that the format of an IO module with only Block readable and no lazy=True argument allowed, similar to StimFitIO, is the appropriate way to achieve this, rather than the RawIO API.

If you are happy with the approach, I will start again and provide an IO class only based on StimfitIO. I believe in this case it would also be appropriate to return the data scaled and so solve the issue discussed below.